app/scripts/models/auth_brokers/fx-ios-v2.js

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * The auth broker to coordinate authenticating for Sync when
+ * embedded in the Firefox for iOS 2.0.
+ */
+
+define([
+  'lib/auth-errors',
+  'models/auth_brokers/fx-desktop-v1'
+], function (AuthErrors, FxDesktopV1AuthenticationBroker) {
+  'use strict';
+
+  var proto = FxDesktopV1AuthenticationBroker.prototype;
+
+  var FxiOSV2AuthenticationBroker = FxDesktopV1AuthenticationBroker.extend({
```

Comment:
I kind of expected this. I've been thinking about how auth broker responsibilities and how we could use FxiOSV2AuthBroker with either WebChannels or CustomEvents. Essentially, we want to use the same behaviors/messaging protocol using two different transports. An auth broker has capabilities, behaviors, communication with the relier, commands, etc, they do a lot and are turning into so called God modules. Channels are already split off, but an auth broker creates them. Maybe we should allow channels to be passed in, and possibly the command set to use too. Or perhaps the commands live with channels and a channel is created for a each transport/command combination.

Something to think about.